src/test/org/pantsbuild/jarjar/integration/JavaVersionsTest.java

```diff
@@ -0,0 +1,74 @@
+package org.pantsbuild.jarjar.integration;
+
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+public class JavaVersionsTest extends IntegrationTestBase {
+
+  @Test
+  public void testShadeJava8Class() throws Exception {
+    testVersion("1.8");
+  }
+
+  @Test
+  public void testShadeJava7Class() throws Exception {
+    testVersion("1.7");
+  }
+
+  @Test
+  public void testShadeJava6Class() throws Exception {
+    testVersion("1.6");
+  }
+
+  private void testVersion(String version) throws Exception {
+    if (!javaVersionIsAtLeast(version)) {
+      System.out.println("Cannot test java " + version + " if this is being run in less than " + version + ".");
+      return;
+    }
+
+    File jar = createJarWithSingleClass(version);
+    List<String> entries = getJarEntries(jar);
+    assertTrue("SingleClass didn't make it into unshaded jar.åå",
```

Comment:
Heh that's what happens when I try to use my tmux prefix, but fail because (1) intellij has focus, and (2) I hit alt instead of control x-)